src/client/sandbox/node/attributes/index.js

```diff
@@ -0,0 +1,76 @@
+import AttributesWrapper from './wrapper';
+import { isHammerheadAttr } from '../../../utils/dom';
+import nativeMethods from '../../native-methods';
+import DomProcessor from '../../../../processing/dom';
+
+const ELEMENT_ATTRIBUTE_WRAPPERS_PROP = 'hammerhead|element-attribute-wrappers-prop';
+
+function cleanAttributes (wrapper) {
+    if (wrapper.length) {
+        for (let i = 0; i < wrapper.length; i++) {
+            delete wrapper[wrapper[i].name];
+            delete wrapper[i];
+        }
+    }
+}
+
+export function assignAttributes (wrapper, attributes) {
+    let length       = 0;
+    const properties = {};
+
+    for (let attr of attributes) {
+        if (!isHammerheadAttr(attr.name)) {
+            const storedAttr = attributes[DomProcessor.getStoredAttrName(attr.name)];
+
+            if (storedAttr) {
+                // eslint-disable-next-line no-restricted-properties
+                if (DomProcessor.isAddedAutocompleteAttr(attr.name, storedAttr.value))
+                    continue;
+
+                attr = nativeMethods.cloneNode.call(attr);
+
+                // eslint-disable-next-line no-restricted-properties
+                attr.value = storedAttr.value;
+            }
+
+            properties[attr.name] = { value: attr, configurable: true };
+            properties[length]    = { value: attr, configurable: true };
+            length++;
+        }
+    }
+
+    properties['length'] = { value: length, configurable: true };
+
+    nativeMethods.objectDefineProperties.call(window.Object, wrapper, properties);
+}
+
+export function getAttributesProperty (el) {
+    if (el[ELEMENT_ATTRIBUTE_WRAPPERS_PROP])
+        return el[ELEMENT_ATTRIBUTE_WRAPPERS_PROP];
+
+    const attributes = nativeMethods.elementAttributesGetter.call(el);
+
+    if (!attributes)
```

Comment:
I guess this line is unnecessary.
If element does not contain attributes then it return an empty NodeNamedMap.
It there are cases when attributes is null or undefined add test cases for it.

Reply:
Ok. Add this to test cases